server.go

```diff
@@ -432,7 +432,7 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB, cc *chainControl,
 	// If we were requested to route connections through Tor and to
 	// automatically create an onion service, we'll initiate our Tor
 	// controller and establish a connection to the Tor server.
-	if cfg.Tor.Active {
+	if cfg.Tor.Active && (cfg.Tor.V2 || cfg.Tor.V3) {
```

Comment:
what if `disableListen` is set?

Reply:
It will also be set `true` here: https://github.com/lightningnetwork/lnd/blob/master/config.go#L495

Reply:
My point is, when `disableListen` is set, you can be sure both `tor.v2` and `tor.v3` are off.

In the code you linked, `tor.v2` and `tor.v3` being off are a condition to set `disableListen`.

Reply:
Ok, so we could do
```go
if cfg.Tor.Active && !cfg.DisableListen {
```
? 😄

Reply:
AFAICT, the result would be exactly the same!

Since this condition determines whether or not we should start the Tor controller (to automatically create a v2/v3 hidden service), I feel like the other is clearer 😸